src/algebra/module/linear_map.lean

```diff
@@ -210,6 +210,63 @@ def comp : M →ₗ[R] M₃ := ⟨f ∘ g, by simp, by simp⟩
 @[norm_cast]
 lemma comp_coe : (f : M₂ → M₃) ∘ (g : M → M₂) = f.comp g := rfl
 
+@[simp] theorem comp_id : f.comp id = f :=
+linear_map.ext $ λ x, rfl
+
+@[simp] theorem id_comp : id.comp f = f :=
+linear_map.ext $ λ x, rfl
+
+end
+
+section
+
+variables [semimodule R M] (f : M →ₗ[R] M)
+
+/-- Iteration of a linear map is a linear map -/
+def iterate : ℕ → (M →ₗ[R] M)
```

Comment:
How reasonable is it to define this as `(\lam g, g.comp f)^[n]`?

Reply:
I pushed what I think you meant; I don't feel strongly either way.

Reply:
Sorry for being unclear, I meant
```lean
def iterate (n : ℕ) (f : M →ₗ[R] M) : M →ₗ[R] M :=
(λ g, linear_map.comp g f)^[n] linear_map.id
```

Reply:
But your new version has the nice benefit of making `iterate_to_fun` a rfl lemma

Reply:
Oh, I see. I have no preference